File: frp/coroutine.py
# -*- coding: utf8 -*-

def foo():
  for i in range(10):
    yield i
    print u'foo: Coroutine'

bar = foo()

print bar.next()
print u'main: MainThread'
print bar.next()
print u'main: MainThread: hello world'

# @references http://blog.ez2learn.com/2010/07/17/talk-about-coroutine-and-gevent/

File: frp/coroutine.lua

function foo()
  print('hello')
  coroutine.yield()
  print('world')
end

local bar = coroutine.create(foo)

coroutine.resume(bar)
print('main, not in `foo()')
coroutine.resume(bar)

-- @https://electronic-blue.herokuapp.com/blog/2012/06/coroutine-an-introduction/
